tests/src/test/java/alluxio/proxy/s3/S3ClientRestApiTest.java

```diff
@@ -205,6 +209,173 @@ public void deleteNonEmptyBucket() throws Exception {
   }
 
   @Test
+  public void getBucket() throws Exception {
+    final String bucket = "bucket-to-get";
+    AlluxioURI uri = new AlluxioURI(AlluxioURI.SEPARATOR + bucket + AlluxioURI.SEPARATOR);
+    new TestCase(mHostname, mPort, S3_SERVICE_PREFIX + AlluxioURI.SEPARATOR + bucket, NO_PARAMS,
+        HttpMethod.PUT, null, TestCaseOptions.defaults()).run();
+    // Verify the directory is created for the new bucket.
+    Assert.assertTrue(mFileSystemMaster.listStatus(uri, ListStatusOptions.defaults()).isEmpty());
+
+    // Prepare a bucket with direct child objects and objects within sub directories:
+    // - /file1
+    // - /file2
+    // - /dir1/subdir1/file3
+    // - /dir2/
+    AlluxioURI file1 = new AlluxioURI(uri.getPath() + "/file1");
+    mFileSystemMaster.createFile(file1, CreateFileOptions.defaults());
+    AlluxioURI file2 = new AlluxioURI(uri.getPath() + "/file2");
+    mFileSystemMaster.createFile(file2, CreateFileOptions.defaults());
+    AlluxioURI dir1 = new AlluxioURI(uri.getPath() + "/dir1");
+    mFileSystemMaster.createDirectory(dir1, CreateDirectoryOptions.defaults());
+    AlluxioURI dir2 = new AlluxioURI(uri.getPath() + "/dir2");
+    mFileSystemMaster.createDirectory(dir2, CreateDirectoryOptions.defaults());
+    AlluxioURI subdir1 = new AlluxioURI(uri.getPath() + "/dir1/subdir1");
+    mFileSystemMaster.createDirectory(subdir1, CreateDirectoryOptions.defaults());
+    AlluxioURI file3 = new AlluxioURI(subdir1.getPath() + "/file3");
+    mFileSystemMaster.createFile(file3, CreateFileOptions.defaults());
+
+    // Expected result.
+    List<URIStatus> objectsList = new ArrayList<>();
+    objectsList.add(
+        new URIStatus(mFileSystemMaster.getFileInfo(file1, GetStatusOptions.defaults())));
+    objectsList.add(
+        new URIStatus(mFileSystemMaster.getFileInfo(file2, GetStatusOptions.defaults())));
+    objectsList.add(
+        new URIStatus(mFileSystemMaster.getFileInfo(file3, GetStatusOptions.defaults())));
+    objectsList.add(
+        new URIStatus(mFileSystemMaster.getFileInfo(dir2, GetStatusOptions.defaults())));
+    ListBucketResult expected = new ListBucketResult(
+        AlluxioURI.SEPARATOR + bucket, objectsList, ListBucketOptions.defaults());
+
+    // Verify op with no param
+    new TestCase(mHostname, mPort, S3_SERVICE_PREFIX + AlluxioURI.SEPARATOR + bucket, NO_PARAMS,
+        HttpMethod.GET, expected,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE)).run();
+
+    // Verify op with prefix
```

Comment:
move to an independent test `getBucketWithPrefix`?

Reply:
done